integration-tests/src/tests/client/state_dump.rs

```diff
@@ -120,3 +133,247 @@ fn test_state_dump() {
+                    }
+                }
+            }
+            if all_parts_present {
+                ControlFlow::Break(())
+            } else {
+                ControlFlow::Continue(())
+            }
+        })
+        .await
+        .unwrap();
+
+        // Simulate state sync
+        println!("syncing node: simulating state sync..");
+        env.clients[1].chain.set_state_header(0, sync_hash, state_sync_header).unwrap();
+        let runtime_client_1 = Arc::clone(&env.clients[1].runtime_adapter);
+        let runtime_client_0 = Arc::clone(&env.clients[0].runtime_adapter);
+        let client_0_store = runtime_client_0.store();
+
+        for part_id in 0..num_parts {
+            let key = StatePartKey(sync_hash, 0, part_id).try_to_vec().unwrap();
+            let part = client_0_store.get(DBCol::StateParts, &key).unwrap().unwrap();
+
+            runtime_client_1
+                .apply_state_part(0, &state_root, PartId::new(part_id, num_parts), &part, &epoch_id)
+                .unwrap();
+        }
+        env.clients[1].chain.set_state_finalize(0, sync_hash, Ok(())).unwrap();
+        println!("syncing node: state sync finished.");
+
+        let synced_block = env.clients[1].chain.get_block(&sync_hash).unwrap();
+        let synced_block_header = env.clients[1].chain.get_block_header(&sync_hash).unwrap();
+        let synced_block_tip = Tip::from_header(&synced_block_header);
+        let response = env.clients[1].runtime_adapter.query(
+            ShardUId::single_shard(),
+            &synced_block.chunks()[0].prev_state_root(),
+            synced_block_tip.height,
+            0,
+            &synced_block_tip.prev_block_hash,
+            &synced_block_tip.last_block_hash,
+            synced_block_header.epoch_id(),
+            &QueryRequest::ViewAccount { account_id: "test_account".parse().unwrap() },
+        );
+
+        if is_final_block_in_new_epoch {
+            assert_matches!(
+                response.unwrap().kind,
+                QueryResponseKind::ViewAccount(_),
+                "the synced node should have information about the created account"
+            );
+        } else {
+            assert!(response.is_err());
+            assert_matches!(
+                response.unwrap_err(),
+                QueryError::UnknownAccount { .. },
+                "the synced node should not have information about the created account"
+            );
+        }
+        actix_rt::System::current().stop();
+    });
+}
+
+#[test]
+/// test state sync with state parts that were dumped when the dumping node's head is in new epoch but final block is not
+fn test_state_sync_w_dumped_parts() {
+    let epoch_length = 5;
+    // excluding account_creation_at_epoch_height=1 because first epoch's epoch_id not being block hash of its first block cause issues
+    for account_creation_at_epoch_height in 2..=4 as u64 {
+        println!("account_creation_at_epoch_height = {}", account_creation_at_epoch_height);
```

Comment:
I'm unsure about this, but seems that `tracing::info!()` is preferred. It then can be nicely integrated with `init_test_logging()`.